### ISSUE TITLE
Predictions are in a very narrow range

### ISSUE THREAD

**Issue by jmoch** [chemprop/chemprop#187]
Hi,
I'm trying to predict a dataset in which the training values range from -2 to 1 in log value.  Chemprop predictions from training on this set are only ranging from -1.3 to -0.1.  Our guess is that the NN is getting stuck in some minima.  Have you seen this before and do you have any suggestions on how to tweak or resolve this?

Thank in advance for your Help,
Jason

**Comment by hesther** [comment]
Hi Jason, your prediction range from -1.3 to -0.1 does that refer to predicting on the training set?

If yes: Your model is underfitting quite badly, i.e. your training loss is very high and the predictions will be centred around the mean of your training distribution. This means that either the representation or the model architecture or both have severe limitations in reproducing your data. For example, if you would try to predict the mass of a molecule (an extrinsic property), but would use mean aggregation after the MPNN (default, but only makes sense for intrinsic properties), you would see results just like what you describe. If you are predicting something extrinsic (which depends on the size of the molecule) try using `--aggregation norm`. If you tell me more about what property you are trying to predict I can help you identify more potential remedies. E.g. if you had data of different molecules in different solvents, you would need to include the solvents as features, etc.

If no (the predictions in the range -1.3 to -0.1 refer to the test set): Check whether the distribution of the test set just is that narrow. If not, read the above answer.

Hope this helps!

**Comment by jmoch** [comment]
Hi Hesther,

Thanks for the response!

You are correct.  I am training on data from -2 to 1 but when I build the model and try to predict the training set the predicted values range from -1.3 to -0.1.  I admit that I am trying to predict a difficult endpoint in PK exposure.  I think it's worth trying your suggestion as well as creating some custom descriptors outside of the default RDK set.

Thanks,
Jason

**Comment by hesther** [comment]
Yeah, in that case it looks like either the representation or the model architecture has some severe shortcoming. A few ideas to test/think about:

- is aggregation between MPNN and FFN a good choice (default: mean, worth to try: norm)
- are atom and bond features sufficient (for some problems, the ring size is e.g. an important property to include additionally. Worth to try: Add custom atom/bond features)
- are there any other circumstances influencing your data, e.g. temperatur, solvent, measurement method, etc. Then that would be good to include as custom molecular descriptor. It is furthermore worth to try some of the build-in descriptors, such as rdkit_2d_normalized (but I think you have done that already). You can use both custom and build-in molecular descriptors at the same time.
- is your target controlled by long-range interactions? Then it might be worth to increase the MPNN depth, and/or add custom molecular descriptors.

**Comment by jmoch** [comment]
Thank you Hesther!  I'll take a look a bit deeper.

Jason
